app/models/user_session.rb

```diff
@@ -9,6 +11,47 @@ def initialize(session)
     @session['answers'][Steps::Excise.id] ||= {}
   end
 
+  def self.build_from_params(session, params)
+    import_destination = params[:import_destination]
+    country_of_origin = params[:country_of_origin]
+    service = (import_destination == 'XI' ? 'xi' : 'uk')
+    other_country_of_origin = ''
+
+    # This reflects an idiosyncrasy in the xi_option form for country_of_origin and how we store the value on the session slightly differently. We'd need to refactor the way we access that if we want to remove this and just use country_of_origin as per the gb and non-row ni routes.
+    if import_destination == 'XI' && !Api::GeographicalArea.eu_member?(country_of_origin)
+      other_country_of_origin = country_of_origin
+      country_of_origin = 'OTHER'
+    end
```

Comment:
this logic seems not to belongs here,
to start we could extract this logic to two private methods:

```
private 
other_country_of_origin(import_destination, eu_member)
country_of_origin(import_destination, eu_member)
```

to isolate it, so it'd be easy to move in the appropriate class.

Reply:
This method is about orchestrating the variables that are required for building the user session from the parameters. We'd end up with lots of small methods where the method calls themselves are as big as the method body which also feels wrong to me.

We could isolate the user session builder into a service. Ultimately what it's building though is the user session so it feels appropriate that this goes here

Reply:
I do think there's scope for clearing up the getters/setters on the user session, though. This could simplify the user session abstraction some what